test/scratchbooks/weileratherton.jl

```diff
@@ -0,0 +1,138 @@
+using Meshes
+using CairoMakie
+
+using ScopedValues
+
+t = Triangle((0.0, 0.0), (1.0, 0.0), (0.5, 1.0))
+r = Quadrangle((0.0, 0.0), (1.0, 0.0), (1.0, 1.0), (0.0, 1.0))
+h = PolyArea(
+        rings(r)...,
+        (reverse.(t |> Scale(0.5) |> Translate((0.3, 0.3)) |> rings))...,
+)
+
+c = Ngon(
+    (0.0, 0.0),
+    (1.0, 0.0),
+    (1.0, 0.2),
+    (0.3, 0.2),
+    (0.3, 0.8),
+    (1.0, 0.8),
+    (1.0, 1.0),
+    (0.0, 1.0),
+)
+
+viz(c)
+
+##
+
+plot_count = 0
+
+function clip_and_show(poly, ring, skip_sutherland=false)
+    plot_count = plot_count + 1
+    println("\n\nClipping $plot_count\n")
+    out = clip(poly, ring, WeilerAtherton())
+    println("!!!  $out")
+    f = Figure(size=(1100, 400))
+    f[1, 1] = Axis(f)
+    tr1 = Translate(0.0, 0.0)
+    tr2 = Translate(2.0, 0.0)
+    tr3 = Translate(3.0, 0.0)
+
+    outs = []
+    push!(outs, poly)
+    push!(outs, rings(tr1(ring))[1])
+    if typeof(out) <: GeometrySet
+        for i in 1:nelements(out)
+            push!(outs, tr2(element(out, i)))
+        end
+    elseif !isnothing(out)
+        push!(outs, tr2(out))
+    end
+
+    if !skip_sutherland
+        out2 = clip(poly, ring, SutherlandHodgman())
+        println("!!!  $out2")
+        if !isnothing(out2)
+            push!(outs, tr3(out2))
+        end
+    end
+
+    viz!(GeometrySet(outs); showsegments=false, color=[7, 11, 3, 5, 6, 8])
+    global plot_count
+    save("out_$plot_count.png", f)
+end
+##
+clip_and_show(r, t |> Scale(0.8) |> Translate(0.25, 0.1))
+clip_and_show(r, t |> Scale(0.8) |> Translate(-0.25, 0.1))
+clip_and_show(r, t |> Scale(0.8) |> Translate(0.0, 0.3))
+clip_and_show(r, t |> Scale(0.8) |> Translate(0.0, 0.0))
+clip_and_show(r, t |> Scale(0.8) |> Translate(0.1, 0.0))
+clip_and_show(r, t |> Scale(0.8) |> Translate(0.2, 0.3))
+clip_and_show(r, t |> Scale(0.8) |> Translate(-0.2, 0.3))
+clip_and_show(r, t |> Scale(0.8) |> Translate(-0.2, -0.3))
```

Comment:
Appreciate if you can remove this file from the PR, and add some of the resulting visualizations as comments instead. You can also share this script in the comments for future reproduction.

Reply:
Removed.